packages/core/src/lib/message/version_0.ts

```diff
@@ -125,11 +129,21 @@ export function createEncoder({
   ephemeral,
   metaSetter
 }: EncoderOptions): Encoder {
+  if (pubsubTopicShardInfo) {
+    pubsubTopicShardInfo.shard
+      ? singleShardInfoToPubsubTopic(pubsubTopicShardInfo)
+      : contentTopicToPubsubTopic(contentTopic, pubsubTopicShardInfo.clusterId);
+  }
   return new Encoder(
     contentTopic,
     ephemeral,
     pubsubTopicShardInfo
```

Comment:
This happens in more than 2 places at this point. 
Let's abstract it (e.g utility function or something else)

Reply:
abstracted away in latest commit